src/main/java/math/Matrix4x4.java

```diff
@@ -112,99 +103,106 @@ public Matrix4x4 inverse() throws IllegalArgumentException {
                     cells[3][1], cells[3][2], cells[3][3])
                     * detInv);
-            m.setCell(2, 1, -determinant3x3(
+            m.setCell(2, 1, -MatrixHelper.determinant3x3(
                     cells[0][0], cells[0][2], cells[0][3],
                     cells[1][0], cells[1][2], cells[1][3],
                     cells[3][0], cells[3][2], cells[3][3])
                     * detInv);
-            m.setCell(2, 2, determinant3x3(
+            m.setCell(2, 2, MatrixHelper.determinant3x3(
                     cells[0][0], cells[0][1], cells[0][3],
                     cells[1][0], cells[1][1], cells[1][3],
                     cells[3][0], cells[3][1], cells[3][3])
                     * detInv);
-            m.setCell(2, 3, -determinant3x3(
+            m.setCell(2, 3, -MatrixHelper.determinant3x3(
                     cells[0][0], cells[0][1], cells[0][2],
                     cells[1][0], cells[1][1], cells[1][2],
                     cells[3][0], cells[3][1], cells[3][2])
                     * detInv);
-            m.setCell(3, 0, -determinant3x3(
+            m.setCell(3, 0, -MatrixHelper.determinant3x3(
                     cells[0][1], cells[0][2], cells[0][3],
                     cells[1][1], cells[1][2], cells[1][3],
                     cells[2][1], cells[2][2], cells[2][3])
                     * detInv);
-            m.setCell(3, 1, determinant3x3(
+            m.setCell(3, 1, MatrixHelper.determinant3x3(
                     cells[0][0], cells[0][2], cells[0][3],
                     cells[1][0], cells[1][2], cells[1][3],
                     cells[2][0], cells[2][2], cells[2][3])
                     * detInv);
-            m.setCell(3, 2, -determinant3x3(
+            m.setCell(3, 2, -MatrixHelper.determinant3x3(
                     cells[0][0], cells[0][1], cells[0][3],
                     cells[1][0], cells[1][1], cells[1][3],
                     cells[2][0], cells[2][1], cells[2][3])
                     * detInv);
-            m.setCell(3, 3, determinant3x3(
+            m.setCell(3, 3, MatrixHelper.determinant3x3(
                     cells[0][0], cells[0][1], cells[0][2],
                     cells[1][0], cells[1][1], cells[1][2],
                     cells[2][0], cells[2][1], cells[2][2])
                     * detInv);
         }
-        return m;
-    }
-
-    public boolean equals() {
-        for (int i = 0; i < 4; i++) {
-            for (int j = 0; j < 4; j++) {
-                if (cells[i][j] != cells[i][j]) {
-                    return false;
+        return m.transpose();
+    }
+
+    public boolean equals(Object obj){
+        float epsilon = 0.000001f;
+        if(obj instanceof Matrix4x4){
+            for(int i=0;i<4;i++){
+                for(int j=0;j<4;j++){
+                    if((((Matrix4x4) obj).getCell(i,j)==-0f&&this.getCell(i,j)==0f)||((Matrix4x4) obj).getCell(i,j)==0f&&this.getCell(i,j)==-0f){
+                        //do nothing
+                    }
```

Comment:
??????

Reply:
Clean up if condition's statement. Remove if not necessary or implement the body.

Reply:
Fix & Merge, do not wait for my review, teammates can approve it.